kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java

```diff
@@ -0,0 +1,137 @@
+                .withNewSpec()
+                .withNewResourceAttributes()
+                .withResource("pod")
+                .withResource("pod2")
+                .withVerb("list")
+                .endResourceAttributes()
+                .endSpec().done();
+
+        assertNotNull(reviewResponse);
+        assertTrue(reviewResponse.getStatus().getAllowed());
+    }
+
+    @Test
+    @Order(2)
+    public void testLoadFromFile() {
+        KubernetesClient client = server.getClient();
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+
+        SelfSubjectAccessReview review = null;
+
+        try {
+            review = mapper.readValue(getClass().getResourceAsStream("/test-selfsubjectaccessreview.yml"), SelfSubjectAccessReview.class);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        assertNotNull(review);
+
+        SelfSubjectAccessReview reviewResponse = client.subjectAccessReviewAuth().inAnyNamespace().create(review);
+
+        assertNotNull(reviewResponse);
+        assertTrue(reviewResponse.getStatus().getAllowed());
+    }
+
+    @Test
+    @Order(3)
+    public void testGet() {
+        KubernetesClient client = server.getClient();
+
+        SelfSubjectAccessReview reviewResponse = client.subjectAccessReviewAuth().inAnyNamespace().createNew()
+                .withNewSpec()
+                .withNewResourceAttributes()
+                .withResource("pod")
+                .withVerb("get")
+                .endResourceAttributes()
+                .endSpec().done();
+
+        assertNotNull(reviewResponse);
+        assertTrue(reviewResponse.getStatus().getAllowed());
+    }
+
+    @Test
+    @Order(4)
+    public void testDelete() {
+        KubernetesClient client = server.getClient();
+
+        SelfSubjectAccessReview reviewResponse = client.subjectAccessReviewAuth().inAnyNamespace().createNew()
+                .withNewSpec()
+                .withNewResourceAttributes()
+                .withResource("pod")
+                .withVerb("delete")
+                .endResourceAttributes()
+                .endSpec().done();
+
+        assertNotNull(reviewResponse);
+        assertTrue(reviewResponse.getStatus().getAllowed());
+    }
+
+    @BeforeEach
+    private void initializeCreateRequestHandler() {
+        server.expect().post().withPath("/apis/authorization.k8s.io/v1/namespaces/test/selfsubjectaccessreviews").andReply(200, recordedRequest -> {
+            ObjectMapper mapper = new ObjectMapper();
+
+            SelfSubjectAccessReview reviewResponse = null;
+
+            try {
+                reviewResponse = mapper.readValue(recordedRequest.getBody().readString(Charset.defaultCharset()), SelfSubjectAccessReview.class);
+                reviewResponse.setStatus(new SubjectAccessReviewStatus(true, false, "",""));
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Test should fail if there is an exception, add `throws IOException` to method signature and remove try-catch block.